adafruit_si7021.py

```diff
@@ -19,11 +19,140 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
+
 """
-`adafruit_si7021`
-====================================================
+``adafruit_si7021``
+===================
 
-TODO(description)
+This is a CircuitPython driver for the SI7021 temperature and humidity sensor.
 
-* Author(s): Radomir Dopieralski
 """
+
+
+from micropython import const
+import ustruct
+import sys
+
+from adafruit_bus_device.i2c_device import I2CDevice
+
+
+HUMIDITY = const(0xf5)
+TEMPERATURE = const(0xf3)
+_RESET = const(0xfe)
+_READ_USER1 = const(0xe7)
+_USER1_VAL = const(0x3a)
+
+
+def _crc(data):
+    crc = 0
+    for byte in data:
+        crc ^= byte
+        for i in range(8):
+            if crc & 0x80:
+                crc <<= 1
+                crc ^= 0x131
+            else:
+                crc <<= 1
+    return crc
+
+
+class SI7021:
+    """
+    A driver for the SI7021 temperature and humidity sensor.
+    """
+
+    def __init__(self, i2c, address=0x40):
+        self.i2c_device = I2CDevice(i2c, address)
+        self.init()
+        self._measurement = 0
+
+    def init(self):
+        self.reset()
+        # Make sure the USER1 settings are correct.
+        while True:
+            # While restarting, the sensor doesn't respond to reads or writes.
+            try:
+                data = bytearray([_READ_USER1])
+                with self.i2c_device as i2c:
+                    i2c.writeto(data, stop=False)
+                    i2c.readfrom_into(data)
+                value = data[0]
+            except OSError as e:
+                if e.args[0] not in ('I2C bus error', 19): # errno 19 ENODEV
+                    raise
+            else:
+                break
+        if value != _USER1_VAL:
+            raise RuntimeError("bad USER1 register (%x!=%x)" % (
+                value, _USER1_VAL))
+
+    def _command(self, command):
+        with self.i2c_device as i2c:
+            i2c.writeto(ustruct.pack('B', command))
+
+    def _data(self):
+        data = bytearray(3)
+        data[0] = 0xff
+        while True:
+            # While busy, the sensor doesn't respond to reads.
+            try:
+                with self.i2c_device as i2c:
+                    i2c.readfrom_into(data)
+            except OSError as e:
+                if e.args[0] not in ('I2C bus error', 19): # errno 19 ENODEV
+                    raise
+            else:
+                if data[0] != 0xff: # Check if read succeeded.
+                    break
+        value, checksum = ustruct.unpack('>HB', data)
+        if checksum != _crc(data[:2]):
+            raise ValueError("CRC mismatch")
+        return value
+
+    def reset(self):
+        self._command(_RESET)
+
+    @property
+    def relative_humidity(self):
+        """
+        The measured relative humidity in percents.
+
+        If no measurement was started, this will start it automatically.
```

Comment:
I don't think this comment is necessary because you are handling the start of the measurement internally yourself.